api/types.proto

```diff
@@ -713,3 +727,31 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+// SecretReference is the linkage between a service and a secret that it uses.
+message SecretReference {
+	// Name is the name of the secret that this references.
+	string name = 1;
+
+	// SecretValueID represents the ID of the specific Secret version that we're
+	// referencing. This identifier exists so that SecretReferences don't leak
+	// any information about the secret contents.
+	string secret_value_id = 2 [(gogoproto.customname) = "SecretValueID"];
```

Comment:
@diogomonica Should we then remove secret version limits, (e.g. how many versions one can have per name), and instead just have a limit on the total number of secrets?  

In such a scenario, what does `UpdateSecret` do?  If the actual secret data has changed, do we just create a new `Secret` with a new ID?  (e.g. it becomes update or create)